### PR TITLE
refs #525 Support for HTML5 multiple file upload input element

### PR DESCRIPTION
**Pull request by albuquerque**
Added support for html5 input element with multiple file upload support.

Code derived from https://github.com/n1k0/casperjs/pull/526.

Tested with phantomjs 2.0.0

### CODE REVIEW

File: modules/casper.js [hunk 1]
```diff
@@ -821,16 +821,19 @@ Casper.prototype.fillForm = function fillForm(selector, vals, options) {
                 if (!file || !file.path) {
                     return;
                 }
-                if (!fs.exists(file.path)) {
-                    throw new CasperError('Cannot upload nonexistent file: ' + file.path);
-                }
+                var paths = (utils.isArray(file.path) && file.path.length > 1) ? file.path :[file.path];
```
Comment (paazmaya) [inline modules/casper.js]: When `file.path` is array with one or zero items, the `paths` will be an array containing an array.

File: modules/casper.js [hunk 2]
```diff
@@ -821,16 +821,19 @@ Casper.prototype.fillForm = function fillForm(selector, vals, options) {
...
+                            }
+                        },this);
                 var fileFieldSelector;
                 if (file.type === "names") {
                     fileFieldSelector = [selector, 'input[name="' + file.selector + '"]'].join(' ');
                 } else if (file.type === "css" || file.type === "labels") {
                     fileFieldSelector = [selector, file.selector].join(' ');
                 }
-                this.page.uploadFile(fileFieldSelector, file.path);
+                this.page.uploadFile(fileFieldSelector,paths);
```
Comment (paazmaya) [inline modules/casper.js]: White space missing after colon

**Comment by paazmaya** [comment]
All PhantomJS related tests seem to fail...

File: tests/suites/casper/formfill.js [hunk 3]
```diff
@@ -219,7 +236,7 @@ casper.test.begin('getFormValues() tests', 2, function(test) {
     var fpath = fs.pathJoin(phantom.casperPath, 'README.md');
     var fileValue = 'README.md';
     if (phantom.casperEngine === 'phantomjs') {
-        if (utils.matchEngine({
+        if (!utils.matchEngine({
             name: 'phantomjs',
```
Comment (istr) [inline tests/suites/casper/formfill.js]: Probably this change is why all phantomjs tests fail. Maybe you should instead just drop the "max" value and test with 2.1.1 as well.

Comment (albuquerque) [inline tests/suites/casper/formfill.js]: Reverted this change. I have left max in there. The getFormValues() fakepath bug seems to  affect only 2.0.0. So now we skip the test for "file multiple" in 2.0.0 similar to other file tests.

**Comment by albuquerque** [comment]
Can #526 be closed as this is derived from the same code as it? 
I have made the requested changes and the tests are now passing.

**Comment by albuquerque** [comment]
Closing this pull request as this was created on master branch.

**Comment by addedjacky** [comment]
use phantomjs 2.1.1

**Pull request closed without merging**